### ISSUE TITLE
API call to /usergroups  - Create user group should not fail with a 500 error

### ISSUE THREAD

**Issue by Vad1mo** [goharbor/harbor#14621]
When trying to create a user group that already exists, harbor should not respond with an 500 error message and instead use a 409 to expose that an entry already exists.

```sh
2021-04-12T22:50:42Z [ERROR] [/lib/http/error.go:54]: {"errors":[{"code":"INTERNAL_SERVER_ERROR","message":"error occurred in add user group, error: pq: duplicate key value violates unique constraint \"unique_group_name\""}]}
/harbor/src/lib/http/error.go:82, github.com/goharbor/harbor/src/lib/http.apiError
/harbor/src/lib/http/error.go:51, github.com/goharbor/harbor/src/lib/http.SendError
/harbor/src/common/api/base.go:73, github.com/goharbor/harbor/src/common/api.(*BaseAPI).RenderError
/harbor/src/common/api/base.go:234, github.com/goharbor/harbor/src/common/api.(*BaseAPI).SendInternalServerError
/harbor/src/core/api/usergroup.go:157, github.com/goharbor/harbor/src/core/api.(*UserGroupAPI).Post
/harbor/src/vendor/github.com/astaxie/beego/router.go:835, github.com/astaxie/beego.(*ControllerRegister).ServeHTTP
```

Tested on Harbor 2.2.0 and 2.2.1

**Comment by AllForNothing** [comment]
@stonezdj same issue for adding an unexciting user to project member list:
![image](https://user-images.githubusercontent.com/30999793/114831495-079f4880-9e00-11eb-82ea-45af38757a4f.png)

**Comment by stonezdj** [comment]
It is an error in the query condition caused this issue:
https://github.com/goharbor/harbor/blob/release-2.2.0/src/common/dao/group/usergroup.go#L34

It is already fixed in master branch
https://github.com/goharbor/harbor/blob/ac5e908597f6fa3b2dec05d3c3e2290f1e0827b0/src/pkg/usergroup/manager.go#L62

**Comment by Vad1mo** [comment]
cool,  it's fixed by #14413  then.
